File: 12_clousers.js
const parentFunction = (name, coins) => {
  // let coins = 3
  const playGame = () => {
    coins -= 1;
    if (coins > 1) {
      console.log(name + " have " + coins + " coins left");
    } else if (coins == 1) {
      console.log(name + " have " + coins + " coin left");
    } else {
      console.log(name + " have no coins left");
    }
  };

  return playGame;
};

const tommy = parentFunction("mohammad", 3);
const jenny = parentFunction("jenny", 5);
tommy();
tommy();
jenny();
jenny();

File: 12_clousers.py
def parent_function(person, coins):

    def play_game():
        nonlocal coins
        coins -= 1
        if coins > 1:
            print(person + " have " + str(coins) + " coins left")
        elif coins == 1:
            print(person + " have " + str(coins) + " coin left")
        else:
            print(person + " have no coins left")

    return play_game


tommy = parent_function("mohammad" ,3)
jenny = parent_function("jenny",  4)
tommy() # mohammad have 3 coins left
tommy() # mohammad have 2 coins left
jenny() # jenny have 4 coins left
